src/lib/codap-helper.ts

```diff
@@ -152,6 +162,100 @@ export class CodapHelper {
     await this.configureUserCase(dataContextName, personalDataLabel);
   }
 
+  /**
+   * Takes an existing dataContext and an incoming shared dataContext, and performs the necessary CODAP
+   * requests to transform the existing to match the incoming.
+   *
+   * In particular, we need to account for the case where both tables may contain an attribute with the
+   * same name, but the latter has been moved to a new/different collection.
+   *
+   * If there are any attributes and collections in the existing dataContext that are not in the shares,
+   * this will leave them alone, thus merging both dataContexts together.
+   *
+   * [Upcoming, this this function will take a "delete" flag that will delete collections/attributes in
+   * the existing DC that aren't in the new, for synchronization between two already-shared DCs]
+   */
+  static async mergeDataContexts(existingDataContextName: string, sharedDataContext: DataContext) {
+    const dataContext = await this.getDataContext(existingDataContextName);
+    if (dataContext) {
+      // first run through both DCs and gather the attribute details for each
+      const originalAttributes = [] as Array<{name: string; collection: string; index: number; attr: any }>;
+      const sharedAttributes = [] as Array<{name: string; collection: string; index: number; attr: any; }>;
```

Comment:
I'd suggest a named type here.

Reply:
Agree.

I with there was a concept for a scoped type, that was only defined within the block it was being used. I hate naming things that are only used on in one tiny place. (I get that if I don't export it it's scoped to this module, but it still seems too big.)